### PR TITLE
add parameters to dpnp.prod

### PR DESCRIPTION
**Pull request by shssf**

### CODE REVIEW

File: dpnp/backend/kernels/dpnp_krnl_reduction.cpp [hunk 1]
```diff
@@ -114,47 +114,90 @@ void dpnp_sum_c(void* result_out,
...
     {
         return;
     }
 
-    _DataType* array_1 = reinterpret_cast<_DataType*>(array1_in);
-    _DataType* result = reinterpret_cast<_DataType*>(result1);
+    const _DataType_output init = get_initial_value<_DataType_output>(initial, 1);
+
+    _DataType_input* input = get_array_ptr<_DataType_input>(input_in);
+    _DataType_output* result = get_array_ptr<_DataType_output>(result_out);
```
Comment (densmirn) [inline dpnp/backend/kernels/dpnp_krnl_reduction.cpp:148]: We can avoid `const_cast` for `result_out`. Wouldn't we like to move `get_array_ptr` to utils maybe in separate PR?

File: dpnp/dpnp_iface_mathematical.py [hunk 2]
```diff
@@ -93,6 +93,15 @@
 ]
 
 
+def convert_result_scalar(result, keepdims):
```
Comment (densmirn) [inline dpnp/dpnp_iface_mathematical.py:96]: I would add `to` or `2` to the function name, `convert_result_to_scalar` or `convert_result2scalar`.

**Review by densmirn — APPROVED** [review]
Good code!

**Review by Rubtsowa — APPROVED** [review]

File: dpnp/backend/kernels/dpnp_krnl_reduction.cpp [hunk 3]
```diff
@@ -114,47 +114,90 @@ void dpnp_sum_c(void* result_out,
...
+    if ((input_in == nullptr) || (result_out == nullptr))
     {
         return;
     }
 
-    _DataType* array_1 = reinterpret_cast<_DataType*>(array1_in);
-    _DataType* result = reinterpret_cast<_DataType*>(result1);
+    const _DataType_output init = get_initial_value<_DataType_output>(initial, 1);
+
+    _DataType_input* input = get_array_ptr<_DataType_input>(input_in);
```
Comment (samir-nasibli) [inline dpnp/backend/kernels/dpnp_krnl_reduction.cpp:147]: casting const pointer to non-const?

**Review by samir-nasibli — APPROVED** [review]
LGTM.

File: dpnp/backend/kernels/dpnp_krnl_reduction.cpp (hunk 1 above)
Comment (shssf) [reply dpnp/backend/kernels/dpnp_krnl_reduction.cpp:148]: "yes" and "yes" :-)
`const_cast` for `result` has no meaning but I use the function `get_array_ptr` to consistency.

File: dpnp/dpnp_iface_mathematical.py (hunk 2 above)
Comment (shssf) [reply dpnp/dpnp_iface_mathematical.py:96]: I would vote for `convert_result2scalar` :-)

File: dpnp/backend/kernels/dpnp_krnl_reduction.cpp (hunk 3 above)
Comment (shssf) [reply dpnp/backend/kernels/dpnp_krnl_reduction.cpp:147]: yes. I understand It is bad. I didn't find a method how to use `const ptr` for MKL and PSTL usage

**Pull request merged**